hledger-lib/Hledger/Reports/ReportOptions.hs

```diff
@@ -719,227 +692,95 @@ setWithReport l = overWithReport l . const
 type ReportableLens' s a = forall f. Reportable f String => (a -> f a) -> s -> f s
 
 -- | Lenses for ReportOpts.
--- Note that some of these are not true lenses, as they have a further restriction on
+
```

Comment:
This comment refers to older code, but: in overWithReport and setWithReport, is "Report" referring to their ability to report an error message ? I found it a bit confusing to understand what's going on there, thinking it might be related to ReportOptions or ReportSpec. If so, would overOrError and setOrError be any clearer ?

Reply:
Yes, Report is referring to their ability to report an error message.

Perhaps `overOrEither` since they return an `Either`, rather than throwing an error.